neuralpp/test/quick_tests/symoblic/simple_test.py

```diff
@@ -83,11 +83,16 @@ def test_substitution():
 
 def test_eval():
     """ Test numerical evaluation, including a quantifier example. """
-    x, i = symbols("x i")
+    x, i, k = symbols("x i k")
     assert sqrt(8) != math.sqrt(8)
     assert sqrt(8).evalf() == math.sqrt(8)
     assert cos(2*x).evalf(subs={x: 2.4}) == math.cos(2*2.4)
 
     expr = Sum(Indexed('x', i), (i, 0, 3))  # expr is a quantified expression
-    expr_as_func = lambdify(x, expr)
+    expr_as_func = lambdify(x, expr)  # lambdify() turns the expression into a python function
     assert expr_as_func([1, 2, 3, 4, 5]) == 10
+
+    # doit() can also do computation/simplification, note the interval can be infinite
```

Comment:
What is the different between doit and simplification? When should we use one or the other?

Reply:
I should have avoid using the word "simplification". A better word would be "evaluation". Since the expression is symbolic, `Sum` is just `Sum`, they don't realy "do" the sum. Calling `doit()` would actually do the sum.

So calling `doit()` makes sense on the expressions that you can "do", such as `Sum` and `Integral`. To use another example from the doc, `2*Integral(x, x)` is just `2*Integral(x, x)`, but `(2*Integral(x, x)).doit()` would do the integral and result in `x**2`.

Reply:
Can you please add some of that explanation to the comments? Thanks.

Reply:
Refactored test for `Sum()` and moved test of `doit()` there;
switched to the term "symbolic evaluation" in the comment.